### PR TITLE
feat(cli): build init onboarding TUI — wizard, AI build-debug, adaptive sizing, robust log viewer

### PR DESCRIPTION
**Pull request by WcaleNieWolny**
## What

Adds the full **`capgo build init` onboarding TUI** — an Ink/React terminal wizard that walks a user from zero to a first cloud build for iOS and Android, including an on-failure AI build-debug flow. This replaces the previous narrow PR (#2328) and presents the whole feature as a single squashed commit.

## Why

`build init` previously dropped users into raw prompts. This gives a guided, resumable, full-screen wizard that detects existing state, collects only what's missing, and streams the cloud build — with graceful handling of small terminals and long/garbled remote logs.

## What's included

**Onboarding wizard**
- Platform-picker shell (`OnboardingShell`) rendered once in the alternate-screen buffer; platform chosen up front, then the per-platform app mounts inline.
- iOS: credential detection, App Store Connect API key entry, certificate/provisioning-profile import, duplicate-profile + cert-limit handling.
- Android: keystore (existing/new), GCP service account + Play developer setup.
- Shared: CI-secrets detection/upload (GitHub/GitLab), GitHub Actions workflow, build request + streaming.

**On-failure AI build debug**
- Offers an AI diagnosis when a build fails; renders a fullscreen, scrollable analysis viewer with bidirectional inline↔scroll routing based on terminal height.

**Adaptive terminal sizing**
- Per-platform minimum size enforced *after* the platform is chosen: **iOS 80×38**, **Android 80×49** (Android's GCP explainer is the tall pole; iOS has no Google step). The picker itself only needs the banner floor (44×11) so users can always choose first.
- Resize-reactive: the size is read directly from `stdout` each render (no one-frame lag), and the resize prompt names the floor of the screen actually shown.
- The unbounded iOS **error screen** routes through the scroll viewer when it's taller than the viewport, so the Try-again / Restart / Exit actions are never clipped.

**Robust build-log viewer**
- Sanitizes the remote fastlane stream: splits bare `\r` (in-place "Cruising 🚗" redraws), strips ANSI/escape/C0 control bytes (the viewer re-applies its own ✔/✖/⚠ colouring).
- Renders **one row per line** (truncate, no wrap) so a multi-KB base64 env blob can't balloon the viewport or desync Ink's clear math.
- `less +F`-style follow mode + live elapsed timer.

**Tests** — a VT test harness built on `@xterm/headless` (renders real Ink output through a real terminal grid) plus ~20 test files: frame-fit per step, per-platform min-size, size gates, platform layout, AI fit, build-log sanitize, build-output viewport.

## Test plan

- [ ] `bun run cli:lint`
- [ ] `bun run cli:typecheck`
- [ ] `bun install --frozen-lockfile` (CLI) — lockfile in sync
- [ ] `bun run cli:test` (onboarding + frame-fit + size + sanitize suites)
- [ ] Manual: `capgo build init` on iOS and Android at small/large/ resized terminals
- [ ] Manual: trigger a build failure → AI debug offer → scrollable analysis

## ⚠️ Merge note

This branch was squashed from a point before main landed its **tracking-v2 (actor-scoped) / usage-analytics / hide-AI-model / GitHub-Actions-workflow** changes, which touch the same onboarding files. Those conflicts (`command.ts`, `ui/app.tsx`, `android/ui/app.tsx`, `components.tsx`, `package.json`, `bun.lock`, `tsconfig.json`) must be resolved at merge time — keeping **both** main's tracking-v2 calls and this PR's onboarding rewrite, and unioning the `test:*` scripts. The config-file conflicts are mechanical (newer versions + `@xterm/headless` + script union); the TSX ones graft main's tracking calls into the rewritten flow.

## Summary by CodeRabbit

* **New Features**
  * AI-powered build analysis with inline/fullscreen review, retry/skip and a caller-driven analysis mode
  * Responsive onboarding wizard with platform picker, componentized steps, and a completed-steps summary
  * Fullscreen, scrollable viewers for build output and AI results; adaptive routing based on terminal size
  * Terminal-size gating with clear "too small" prompt

* **Bug Fixes**
  * Sanitized streamed build logs to avoid garbled/fused rows and remove terminal control bytes

* **Chores**
  * CLI version bumped to 7.119.2; added headless terminal helper and expanded automated tests and validation scripts

### CODE REVIEW

File: cli/test/test-frame-fit-log.mjs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    const { visible } = capLogRows(longLog, budget)
+    const cappedEntries = [{ text: '…and N earlier steps done (resize taller to see all)', color: 'gray' }, ...visible]
+    const capped = frameRows(frame(cappedEntries, rows, bodyHeight), 80)
+    assert(capped <= rows, `capped frame is ${capped} rows, exceeds ${rows}`)
+  })
+}
+
+test('UNCAPPED long log overflows — proving the cap is what prevents too-small', () => {
+  const rows = 19
+  const uncapped = frameRows(frame(longLog, rows, 80, 6), 80)
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/test/test-frame-fit-log.mjs]: Fixed in 409224b1 — changed `frame(longLog, rows, 80, 6)` to `frame(longLog, rows, 6)` so `6` is correctly passed as `bodyHeight` (frame takes 3 params). Restores the intended uncapped-overflow assertion.

File: cli/test/test-frame-fit-log.mjs [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  assert(uncapped > rows, `expected the uncapped 30-entry log to overflow ${rows} rows, got ${uncapped}`)
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/test/test-frame-fit-log.mjs:106]: Fixed in 409224b1 — `frame(longLog, rows, 80, 6)` → `frame(longLog, rows, 6)`; the spurious arg made 80 the bodyHeight. Same fix as the duplicate finding above.

File: cli/test/helpers/onboarding-fixtures.mjs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+// build-complete, adding-platform).
+import React from 'react'
+import * as ks from '../../src/build/onboarding/ui/steps/android-keystore.tsx'
+import * as sa from '../../src/build/onboarding/ui/steps/android-sa-gcp.tsx'
+import * as ash from '../../src/build/onboarding/ui/steps/android-shared.tsx'
+import * as aci from '../../src/build/onboarding/ui/steps/android-ci.tsx'
+import * as cred from '../../src/build/onboarding/ui/steps/ios-credentials.tsx'
+import * as imp from '../../src/build/onboarding/ui/steps/ios-import.tsx'
+import * as ish from '../../src/build/onboarding/ui/steps/ios-shared.tsx'
+import * as cic from '../../src/build/onboarding/ui/steps/ios-ci.tsx'
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/test/helpers/onboarding-fixtures.mjs]: Fixed in 0d68738 — renamed the `aci` alias to `androidCi` (cleared the typos check) and alphabetised the step imports. `noop` is left as-is since `cli/test` isn't linted and cli:lint passes.

File: cli/src/build/onboarding/android/ui/app.tsx [hunk 4]
```diff
@@ -1455,16 +1690,86 @@ const AndroidOnboardingApp: FC<AppProps> = ({ appId, initialProgress, androidDir
...
+  // THIS mounted component so all in-progress state (current step, entered
+  // values, build output) is preserved — a shrink shows the prompt, a re-grow
+  // shows the exact same step. Kept as an early return after all hooks so the
+  // rules of hooks hold. The wizard never clips, and never exits, on resize.
+  if (!terminalFitsOnboarding(terminalCols, terminalRows, 'android'))
+    return <TerminalTooSmallPrompt cols={terminalCols} rows={terminalRows} minRows={ANDROID_MIN_ROWS} />
+
+  if (step === 'requesting-build')
+    return <FullscreenBuildOutput title="Building..." lines={buildOutput} terminalRows={terminalRows} />
+
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/onboarding/android/ui/app.tsx]: Fixed in 409224b1 — moved the requesting-build and AI-scroll fullscreen takeovers BEFORE the min-size gate (matching the iOS app), so a mid-build/mid-AI shrink no longer hides the live view behind the resize prompt. Gate still guards the normal step body.

File: cli/src/build/onboarding/ui/steps/android-ci.tsx [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+// signal and threads `dense={true}` here, collapsing each body to the terse,
+// budget-fitting form (blank lines dropped, the per-provider setup advice
+// reduced to the first entry + a "… +N more" hint, the overwrite key list
+// tailed to the last few + a "… +N more" hint, and the target picker capped via
+// Select's `visibleOptionCount` with a "+N more" hint). `dense` defaults to
+// `false` so a component rendered without the prop (e.g. a test asserting the
+// comfortable form) gets the original look. All props/handlers/behaviour are
+// identical across both modes.
+import type { FC } from 'react'
+import type { CiSecretSetupAdvice, CiSecretTarget } from '../../ci-secrets.js'
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/onboarding/ui/steps/android-ci.tsx]: Fixed in 0d68738 — the unused `CiSecretTarget` import and the dead `CheckingCiSecretsStep`/`UploadingCiSecretsStep` were removed (knip green). Remaining `dense` params were already dropped earlier in fe14f37.

File: cli/src/build/onboarding/ui/steps/android-keystore.tsx [hunk 6]
```diff
@@ -0,0 +1,363 @@
...
+  onBack: () => void
+  dense?: boolean
+}
+
+// Comfortable: the original info Alert + a <Newline/> + four indented,
+// full-sentence bullets in a marginLeft box + a <Newline/> + the Back control.
+// Dense: the Alert/box/blank-lines are dropped in favour of terse single-line
+// bullets so every line stays un-wrapped within the 13-row budget at 60 cols
+// (the original wrapping bullets blew the budget there).
+export const KeystoreExplainerStep: FC<KeystoreExplainerStepProps> = ({ onBack, dense = false }) => {
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/onboarding/ui/steps/android-keystore.tsx]: Addressed earlier (fe14f37) — the unused `dense` params were removed from these keystore steps; cli:lint is clean.

File: cli/src/build/onboarding/ui/steps/android-sa-gcp.tsx [hunk 7]
```diff
@@ -0,0 +1,574 @@
...
+// choices — and the Select keeps its window capped so a long message can't push
+// the control off-screen.
+
+export interface SaJsonValidationFailedStepProps {
+  message: string
+  onChoose: (choice: 'retry' | 'save-anyway' | 'oauth') => void
+  dense?: boolean
+}
+
+export const SaJsonValidationFailedStep: FC<SaJsonValidationFailedStepProps> = ({ message, onChoose, dense = false }) => {
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/onboarding/ui/steps/android-sa-gcp.tsx]: Addressed earlier (fe14f37) — unused `dense`/`detectedCount` removed from the sa-gcp steps; cli:lint clean.

File: cli/src/build/onboarding/ui/steps/ios-ci.tsx [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+// <Newline/>, then the Select. Dense: the blank lines + the per-provider reason
+// message drop so two providers' worth of label + command lines still fits 13
+// rows at 60 cols.
+export interface CiSecretsSetupStepProps {
+  advice: CiSecretSetupAdvice[]
+  dense?: boolean
+  onChange: (value: string) => void
+}
+
+export const CiSecretsSetupStep: FC<CiSecretsSetupStepProps> = ({ advice, dense = false, onChange }) => {
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/onboarding/ui/steps/ios-ci.tsx]: Addressed earlier (fe14f37) — unused `dense` removed from the iOS CI steps; cli:lint clean.

File: cli/src/build/onboarding/ui/steps/ios-import.tsx [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
+        {`No provisioning profile on this Mac is linked to "${identityName}".`}
+      </Alert>
+      <Newline />
+      <Text dimColor>
+        The cert is in your Keychain but the matching profile isn't on disk. Pick a recovery path:
+      </Text>
+      <Newline />
+      <Select options={options} onChange={onChange} />
+    </Box>
+  )
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/onboarding/ui/steps/ios-import.tsx]: Addressed earlier (fe14f37) — unused `dense` removed from the iOS import steps; cli:lint clean.

File: cli/src/build/onboarding/android/ui/app.tsx [hunk 10]
```diff
@@ -19,19 +19,93 @@ import { copyFile, readFile } from 'node:fs/promises'
 import { homedir } from 'node:os'
 import { join, resolve as resolvePath } from 'node:path'
 import process from 'node:process'
-import { Alert, ProgressBar, Select } from '@inkjs/ui'
-import { Box, Newline, Text, useApp, useInput, useStdout } from 'ink'
+import { ProgressBar, Select } from '@inkjs/ui'
+import type { DOMElement } from 'ink'
+import { Box, measureElement, Text, useApp, useInput, useStdout } from 'ink'
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/onboarding/android/ui/app.tsx]: Fixed — unused `Select` import removed and imports de-fragmented in the merge resolution; cli:lint passes 0 errors.

File: cli/src/build/onboarding/build-log.ts [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+// eslint-disable-next-line no-control-regex
+const CSI_RE = /\x1B\[[0-9;?]*[ -/]*[@-~]/g
+// Other escape sequences: ESC + a single intermediate/final byte (OSC start,
+// charset selection, etc.). Stripped defensively so no lone ESC survives.
+// eslint-disable-next-line no-control-regex
+const ESC_RE = /\x1B[@-Z\\-_]/g
+// Remaining C0 control bytes + DEL, EXCLUDING tab (\x09), LF (\x0A) and CR
+// (\x0D) — LF/CR are handled by the split above, tab is legitimate indentation.
+// eslint-disable-next-line no-control-regex
+const CTRL_RE = /[\x00-\x08\x0B\x0C\x0E-\x1F\x7F]/g
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/onboarding/build-log.ts:40]: Acknowledged. The project's eslint config (cli:lint) does not enable the `regexp/*` or `e18e/*` rules, so this isn't a CI blocker and the build/lint are green. Happy to apply the suggested escapes if you'd like them as a style preference — left as-is for now to keep the diff minimal.

File: cli/package.json [hunk 12]
```diff
@@ -49,7 +49,7 @@
   ],
   "engines": {
     "npm": ">=8.0.0",
-    "node": ">=20.0.0"
+    "node": ">=22.0.0"
   },
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/package.json:53]: The Node `>=22` bump and React/ink upgrades came from main's versions during the merge (kept the newer of the two sides), not a new decision in this PR. Flagging for maintainer confirmation rather than reverting unilaterally.

File: cli/package.json [hunk 13]
```diff
@@ -94,20 +94,27 @@
...
...-edge-cases && bun run test:regex && bun run test:upload && bun run test:credentials && bun run test:credentials-validation && bun run test:android-service-account-validation && bun run test:build-zip-filter && bun run test:checksum && bun run test:build-needed && bun run test:ci-prompts && bun run test:ci-secrets && bun run test:posthog-exception && bun run test:build-platform-selection && bun run test:onboarding-recovery && bun run test:onboarding-progress && bun run test:onboarding-run-targets && bun run test:run-device-command && bun run test:init-app-conflict && bun run test:init-guardrails && bun run test:prompt-preferences && bun run test:esm-sdk && bun run test:mcp && bun run test:version-detection && bun run test:platform-paths && bun run test:payload-split && bun run test:manifest-path-encoding && bun run test:macos-signing && bun run test:apple-api-import-helpers && bun run test:ai-log-capture && bun run test:ai-analyze-flow && bun run test:ai-render-markdown && bun run test:ai-onboarding-mode && bun run test:ai-fit && bun run test:platform-layout && bun run test:frame-fit && bun run test:onboarding-min-size && bun run test:min-size-gate && bun run test:shell-size-gate",
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/package.json]: Noted — the long `test` script is inherited from main (unioned with the new size/frame test scripts). Splitting it into grouped scripts is a reasonable follow-up but out of scope for this onboarding PR.

File: cli/src/build/request.ts [hunk 14]
```diff
@@ -1977,8 +1985,30 @@ export async function requestBuildInternal(appId: string, options: BuildRequestO
...
+        // releaseCapturedLogs(jobId) explicitly. Without this, the cleanup
+        // handlers registered above would remove it on process exit before
+        // the caller had a chance to read it.
+        keepPromptFile = true
+        const logsPath = `${process.env.CAPGO_AI_LOG_BASE_DIR || '/tmp/capgo-builds'}/${capturedJobId}.log`
+        aiAnalysisInfo = {
+          jobId: capturedJobId,
+          capturedLogPath: logsPath,
+          ready: true,
+        }
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/request.ts:2010]: Good catch — `getLogCapturePath(capturedJobId)` would de-duplicate this. It's a pre-existing pattern in request.ts (not introduced by this PR's onboarding changes), so leaving it for a focused follow-up to avoid widening this PR's surface.

File: cli/src/build/onboarding/command.ts [hunk 15]
```diff
@@ -93,25 +101,30 @@ export async function onboardingBuilderCommand(options: OnboardingBuilderOptions
...
+    }),
+    { alternateScreen: true },
   )
   await waitUntilExit()
+
+  // Durable breadcrumb in the user's normal terminal flow — the alt buffer
+  // restore wiped the wizard's last frame. Written via process.stdout to
+  // bypass the project-wide no-console lint rule (one-shot UX message, not
+  // application logging).
+  process.stdout.write(`\n✔ Capgo onboarding complete for ${appId}${resolvedPlatform ? ` (${resolvedPlatform})` : ''}.\n`)
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/onboarding/command.ts]: Valid P2 — the completion breadcrumb currently prints on any exit, including cancel paths. Fixing it correctly needs a completion signal plumbed from the app through the shell (only fire after reaching `build-complete`); that touches exit logic in 3 files and is riskier than a CI fix, so I'm deferring it to a focused follow-up rather than rushing it here. Tracked.

File: cli/test/test-ai-fit.mjs [hunk 16]
```diff
@@ -0,0 +1,288 @@
...
+  // Must cover the inline chrome (Header + padding + title + caution + Select)
+  // so we don't route inline when it would overflow, but NOT be so large that
+  // analyses scroll on terminals where they'd fit (the old value of 20 did
+  // exactly that — see git history). 6–14 is the sane band.
+  if (AI_RESULT_CHROME_ROWS < 6 || AI_RESULT_CHROME_ROWS > 14)
+    throw new Error(`chrome reserve out of range: ${AI_RESULT_CHROME_ROWS}`)
+})
+
+test('stripAnsi removes SGR escape codes', () => {
+  const styled = `\x1b[1;36mhello\x1b[0m world`
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/test/test-ai-fit.mjs:47]: cli:lint passes 0 errors and `cli/test` isn't covered by the project eslint config, so the `\x1b`-case and `node:process` nits aren't CI blockers. Left as-is to keep the diff focused on the failing checks.

File: cli/test/test-frame-fit-android-shared.mjs [hunk 17]
```diff
@@ -0,0 +1,289 @@
...
+    console.log(`✔ ${name}`)
+  }
+  catch (error) {
+    failed++
+    console.error(`✖ ${name}\n  ${error.message}`)
+  }
+}
+
+const h = React.createElement
+const noop = () => {}
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/test/test-frame-fit-android-shared.mjs:47]: Same as the other test-file nits — `cli/test` isn't linted by cli:lint (which is green), so the top-level-function / `node:process` style points aren't blocking. Left as-is.

File: cli/test/test-frame-fit-resize.mjs [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+
+let passed = 0
+let failed = 0
+async function test(name, fn) {
+  try { await fn(); passed++; console.log(`✔ ${name}`) }
+  catch (error) { failed++; console.error(`✖ ${name}\n  ${error.message}`) }
+}
+function assert(cond, msg) { if (!cond) throw new Error(msg) }
+const h = React.createElement
+const noop = () => {}
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/test/test-frame-fit-resize.mjs:40]: Style-only and in `cli/test` (not covered by cli:lint, which passes). Left as-is to keep this PR's diff scoped to the real CI failures.

File: cli/test/test-frame-fit-viewer.mjs [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+import { stripAnsi } from '../src/build/onboarding/ai-fit.ts'
+import { FullscreenAiViewer } from '../src/build/onboarding/ui/components.tsx'
+
+let passed = 0
+let failed = 0
+function test(name, fn) {
+  try { fn(); passed++; console.log(`✔ ${name}`) }
+  catch (error) { failed++; console.error(`✖ ${name}\n  ${error.message}`) }
+}
+function assert(cond, msg) { if (!cond) throw new Error(msg) }
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/test/test-frame-fit-viewer.mjs:24]: Same — `cli/test` style nits, not flagged by the project's cli:lint (green). Left as-is.

File: cli/test/test-min-size-gate.mjs [hunk 20]
```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bun
+// Behaviour test for the startup size gate (MinSizeGate).
+//
+// Below the floor it must show a resize prompt (naming the short dimension) and
+// hide the wizard; at/above the floor it must render the wizard content through
+// unchanged. Rendered via the real VT grid so we assert what the terminal shows.
+import { Box, Text } from 'ink'
+import process from 'node:process'
+import React from 'react'
+import { MIN_COLS, MIN_ROWS } from '../src/build/onboarding/min-terminal-size.ts'
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/test/test-min-size-gate.mjs:10]: Import-order nit in a test file; cli:lint doesn't cover `cli/test` and is green. Left as-is.

File: cli/tsconfig.json [hunk 21]
```diff
@@ -4,6 +4,7 @@
     "lib": [
       "es2023"
     ],
+    "ignoreDeprecations": "5.0",
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/tsconfig.json]: Fixed in fe14f37 — `ignoreDeprecations` is now `"6.0"` (bumped for the TS6 toolchain inherited from main). This thread is outdated.

File: cli/package.json (hunk 12 above)
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/package.json:53]: You're right and my earlier reply was wrong — I re-checked: main is `>=20.0.0` and the `>=22.0.0` floor was introduced by this branch (an accidental artifact of merging package.json versions), with no Node-22-only API in cli/src. Reverted to `>=20.0.0` in 87f9cc4 — thanks for catching it.

File: cli/package.json (hunk 13 above)
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/package.json]: No need for a follow-up issue from your side — noted internally; happy to leave the test-script grouping as a future cleanup.

File: cli/src/build/request.ts (hunk 14 above)
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/request.ts:2010]: No follow-up issue needed — I'll keep the `getLogCapturePath` de-dup on the informal cleanup list. Thanks.

File: cli/src/build/onboarding/command.ts (hunk 15 above)
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/onboarding/command.ts]: Fixed in af6ffd4d. Added an OnboardingResult contract: command.ts now defaults to { outcome: 'cancelled' } and each wizard app reports { outcome: 'completed', summary } ONLY when it reaches build-complete. The ✔ success line now prints solely on 'completed' (missing-platform / cancel paths get a neutral 'exited — not completed' line instead). Also addressed the related findings from your review: the completion summary (build URL + generated workflow/.env paths + CI-secret summary) is now carried out via onResult and reprinted to the PRIMARY buffer so it survives the alt-screen teardown; and the cli/package.json version regression (7.119.0 → 7.119.1) is fixed. Thanks for the catches.

File: cli/package.json [hunk 22]
```diff
@@ -105,11 +105,18 @@
...
     "test:ai-analyze-flow": "bun test/test-ai-analyze-flow.mjs",
-    "test:ai-render-markdown": "bun test/test-ai-render-markdown.mjs"
+    "test:ai-render-markdown": "bun test/test-ai-render-markdown.mjs",
+    "test:ai-onboarding-mode": "bun test/test-ai-onboarding-mode.mjs",
+    "test:ai-fit": "bun test/test-ai-fit.mjs",
+    "test:platform-layout": "bun test/test-platform-layout.mjs",
+    "test:frame-fit": "bun test/run-frame-fit.mjs",
+    "test:onboarding-min-size": "bun test/test-onboarding-min-size.mjs",
+    "test:min-size-gate": "bun test/test-min-size-gate.mjs",
+    "test:shell-size-gate": "bun test/test-shell-size-gate.mjs"
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/package.json]: Fixed in 5cd30fb — added `test:build-log-sanitize` + `test:build-output-viewport` scripts and appended both to the aggregate `test` chain. Confirmed they now run (12 + 4 assertions pass). Good catch — these guards weren't actually executing in CI.

File: cli/src/build/onboarding/ui/shell.tsx [hunk 23]
```diff
@@ -0,0 +1,141 @@
...
+  const { cols, rows } = useTerminalSize()
+  const [ready, setReady] = useState<ReadyApp | null>(null)
+
+  // Begin loading the chosen platform's progress; mount the app once it lands.
+  // The picker stays on screen during the (few-ms) load, so there's no loading
+  // frame on the picker path.
+  const choose = useCallback((platform: Platform) => {
+    onResolvePlatform?.(platform)
+    void loadReady(platform, appId).then(setReady)
+  }, [appId, onResolvePlatform])
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/onboarding/ui/shell.tsx]: Fixed in 5cd30fb — `choose()` now `.catch()`es `loadReady`. On failure (e.g. corrupt saved-progress JSON, which `loadProgress` throws on) it renders an error frame, calls `onResult?.({ outcome: 'cancelled' })`, and exits — instead of an unhandled rejection + frozen picker.

File: cli/src/build/onboarding/ui/min-size-gate.tsx [hunk 24]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+// fits ? children : prompt. Use only where unmounting `children` on resize is
+// harmless (no in-progress state). For the stateful wizard apps, render
+// TerminalTooSmallPrompt directly instead (see above).
+export const MinSizeGate: FC<MinSizeGateProps> = ({ cols, rows, children }) => {
+  if (terminalFitsOnboarding(cols, rows))
+    return <>{children}</>
+  return <TerminalTooSmallPrompt cols={cols} rows={rows} />
+}
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/onboarding/ui/min-size-gate.tsx:96]: Fixed in 5cd30fb (doc-only) — updated the MinSizeGate comment to state it gates the full onboarding floor and is NOT used for the picker (the shell gates the picker directly via `terminalFitsPicker` + `TerminalTooSmallPrompt(PICKER_MIN_*)`). MinSizeGate stays for its test + any full-floor caller.

File: cli/src/build/onboarding/ui/steps/ios-credentials.tsx [hunk 25]
```diff
@@ -0,0 +1,446 @@
...
+        <Text bold color="white">Ctrl+O</Text>
+        <Text dimColor> to open App Store Connect in your browser</Text>
+      </Box>
+      <Newline />
+      <Divider />
+      <Newline />
+      {control}
+    </Box>
+  )
+}
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/onboarding/ui/steps/ios-credentials.tsx:224]: Fixed in 5cd30fb (doc-only) — corrected the ApiKeyInstructionsStep comment to describe the actual always-comfortable rendering. The adaptive `dense` collapse was dropped repo-wide (startup size gate guarantees the rows); the `dense` prop is kept only for call-site compatibility. No behavior change intended here.

File: cli/src/build/onboarding/ui/steps/ios-shared.tsx [hunk 26]
```diff
@@ -0,0 +1,577 @@
...
+            </Text>
+            <Newline />
+          </>
+        )}
+        {runHint}
+      </Box>
+      <Newline />
+    </Box>
+  )
+}
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/onboarding/ui/steps/ios-shared.tsx:565]: Fixed in 5cd30fb (doc-only) — corrected the BuildCompleteStep comment: it always renders the comfortable form (paddingY=1 + spacing); the `dense` collapse was dropped with the rest of the dense flag. Prop kept for compatibility only.

File: cli/test/helpers/size-search.mjs [hunk 27]
```diff
@@ -0,0 +1,97 @@
...
+// a size "passes" when no checked frame clips at that size.
+//
+// The search is two independent 1-D searches because cols and rows affect fit
+// almost separably: cols drives text WRAPPING (narrower → taller frames), rows
+// is the vertical budget. We:
+//   1. Pick a generous rows budget, binary-search the minimal COLS at which no
+//      frame's natural width is mangled and wrapping stays bounded.
+//   2. At that cols, find the minimal ROWS = max over frames of naturalRows
+//      (the tallest frame's height) — that's the vertical floor.
+//   3. Verify the (cols, rows) pair passes all checks together.
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/test/helpers/size-search.mjs:15]: Fixed in 5cd30fb (doc-only) — changed 'binary-search' to 'linearly scan upward' to match `minColsAt`/`minRowsAt`, which the inline comment already noted are linear.

File: cli/src/build/request.ts (hunk 14 above)
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/request.ts:2010]: Resolving as won't-fix-in-this-PR: this is a pre-existing pattern in request.ts (not introduced by the onboarding changes) and a non-blocking nitpick. Tracked informally for a focused follow-up to keep this PR's surface scoped.

File: cli/test/test-ai-fit.mjs (hunk 16 above)
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/test/test-ai-fit.mjs:47]: Resolving: per your own learning for this repo, cli/test/** is excluded from cli:lint, so unicorn/escape-case + node:process don't apply here and cli:lint reports 0 errors. No CI impact — leaving the test file as-is.

File: cli/test/test-frame-fit-android-shared.mjs (hunk 17 above)
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/test/test-frame-fit-android-shared.mjs:47]: Resolving: cli/test/** is outside the cli:lint scope (confirmed in your repo learnings), so antfu/top-level-function + node/prefer-global/process are non-blocking here. cli:lint is green. Leaving as-is.

File: cli/src/build/onboarding/ui/steps/ios-credentials.tsx [hunk 28]
```diff
@@ -0,0 +1,446 @@
...
+// (Select) or a direct path input (FilteredTextInput). The submit handler for
+// the no-picker path is owned by the parent. Telemetry/file-reads happen there.
+//
+// Renders the full comfortable form: the info Alert, a <Newline/>, the FOUR
+// numbered setup steps in a marginLeft box, a <Newline/>, the "Press Ctrl+O"
+// hint, a <Newline/>, a Divider, another <Newline/>, then the control (with a
+// <Newline/> before the picker Select). (The startup size gate guarantees the
+// terminal is tall enough, so the old adaptive `dense` collapse was dropped; the
+// `dense` prop is accepted for call-site compatibility but no longer alters the
+// layout — see the per-platform floor in min-terminal-size.ts.)
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/onboarding/ui/steps/ios-credentials.tsx]: You're right — fixed in 20a84b3. My earlier comment over-claimed: ApiKeyInstructionsStep still uses `{!dense && <Newline />}` (line 149) to drop the spacer before the picker Select. Comment now states `dense` only suppresses that single Newline; the multi-step copy is unconditional. (The neighbouring CredentialsExistStep/SetupMethodSelectStep genuinely use dense too — only this one comment was inaccurate.)
